src/layer/track.js

```diff
@@ -0,0 +1,147 @@
+const bridgeSelect = ["==", ["get", "brunnel"], "bridge"];
+const fordSelect = ["==", ["get", "brunnel"], "ford"];
+const notFordSelect = ["!=", ["get", "brunnel"], "ford"];
+const opacity = ["interpolate", ["exponential", 1.2], ["zoom"], 12, 0, 13, 1];
+const getBrunnel = ["get", "brunnel"];
+
+export const track = {
+  id: "highway-track",
+  type: "line",
+  source: "openmaptiles",
+  "source-layer": "transportation",
+  filter: ["all", trackSelect, unpavedSelect],
+  minzoom: 12,
+  paint: {
+    "line-color": ["match", getBrunnel, "ford", Color.waterLine, "#d4b791"],
+    "line-opacity": opacity,
+    "line-blur": 0.75,
+    "line-width": 0.5,
+    "line-dasharray": [12, 3],
+    "line-offset": 0,
+    "line-gap-width": [
+      "interpolate",
+      ["exponential", 1.2],
+      ["zoom"],
+      13,
+      0.7,
+      20,
+      6,
+    ],
+  },
+};
+
+export const pavedTrack = {
+  id: "highway-track-paved",
+  type: "line",
+  source: "openmaptiles",
+  "source-layer": "transportation",
+  filter: ["all", trackSelect, pavedSelect],
+  minzoom: 12,
+  paint: { ...track.paint },
+};
+pavedTrack["paint"]["line-dasharray"] = [1, 0];
+
+export const trackBridge = {
+  id: "highway-track-bridge",
+  type: "line",
+  source: "openmaptiles",
+  "source-layer": "transportation",
+  filter: ["all", trackSelect, unpavedSelect, bridgeSelect],
+  minzoom: 12,
+  paint: { ...track.paint },
+};
+
+export const pavedTrackBridge = {
+  id: "highway-track-paved-bridge",
+  type: "line",
+  source: "openmaptiles",
+  "source-layer": "transportation",
+  filter: ["all", trackSelect, pavedSelect, bridgeSelect],
+  minzoom: 12,
+  paint: { ...pavedTrack.paint },
+};
+
+// Bridge casing layers
+export const bridgeCasing = {
+  id: "track-bridge-casing",
+  type: "line",
+  source: "openmaptiles",
+  "source-layer": "transportation",
+  filter: ["all", bridgeSelect, trackSelect],
+  minzoom: 13,
+  layout: {
+    "line-cap": "butt",
+    "line-join": "bevel",
+    visibility: "visible",
+  },
+  paint: {
+    "line-color": "black",
+    "line-opacity": opacity,
+    "line-width": [
+      "interpolate",
+      ["exponential", 1.2],
+      ["zoom"],
+      13,
+      1.1,
+      20,
+      11,
+    ],
+  },
+};
+// Bridge casing layers
+export const bridgeFill = {
+  id: "track-bridge-fill",
+  type: "line",
+  source: "openmaptiles",
+  "source-layer": "transportation",
+  filter: ["all", bridgeSelect, trackSelect],
+  minzoom: 13,
+  layout: {
+    "line-cap": "butt",
+    "line-join": "bevel",
+    visibility: "visible",
+  },
+  paint: {
+    "line-color": Color.backgroundFill,
+    "line-opacity": opacity,
+    "line-width": [
+      "interpolate",
+      ["exponential", 1.2],
+      ["zoom"],
+      13,
+      1.0,
+      20,
+      10,
+    ],
+  },
+};
+
+export const legendEntries = [
+  {
+    description: "Land access track",
+    layers: [track.id],
+    filter: notFordSelect,
+  },
+  {
+    description: "Land access track - ford",
```

Comment:
So far we’ve omitted brunnels from the legend because taking them out of context can be confusing. (A road bridge, for one thing, looks just like a normal road on its own.) We’d need to add some functionality to the legend to place another linear feature across the spotlighted feature. But I think the ford treatment you’ve implemented looks intuitive enough on its own.

Reply:
For now I've left the ford-specific legend entries to avoid the case of ford-styling ending up in the legend when both unpaved ford and non-ford are present (such as at [this location](http://localhost:1776/#map=16.45/43.086983/-71.069226)) as well as allow a concrete ford surrounded by unpaved tracks to get an entry noting that it is paved (such as at [this location](http://localhost:1776/#map=17.98/54.382105/-1.104678)). Maybe this isn't ideal, but dropping any filters and sometimes showing blue for tracks felt particularly wrong.

Reply:
I certainly agree with filtering out fords from the general track entries, but in the absence of the crossing lines capability I described above, it’s probably less confusing overall to omit any entries about the fords.

The main issue is actually that the legend is depicting the track as a single dashed line, whereas it’s actually two parallel lines. I assume that’s because it doesn’t honor the `line-offset` and `line-gap-width` properties. If we fix that, then at least the ford entry won’t look like an intermittent creek.

If we keep the ford entries around, we might be able to combine them into one entry. There are other examples where we avoided breaking out combinations of attributes – for example, there’s only one expressway entry, even though multiple classes of roads that render differently can be expressways. This works because the entry clearly is about a particular aspect of the line, not trying to visually represent every possible expressway.

Reply:
#770 tracks the incorrect implementation of `line-gap-width` in the legend.